analysis_request_cli/create_request.py

```diff
@@ -0,0 +1,204 @@
+            msg = 'Created assemblyJob for run {}'
+        logging.info(msg.format(run.primary_accession))
+
+
+def create_new_assembly_annotation_jobs(ena, mh, study, request, args):
+    # TODO implement to support studies containing assemblies
+    assemblies = ena.get_study_assembly_accessions(study.secondary_accession, False, args.private)
+    if not len(assemblies):
+        logging.warning('No runs to annotate in this study.')
+        return
+    annotated_assemblies = mh.get_up_to_date_annotation_jobs(study.secondary_accession)
+    assemblies = filter_duplicate_runs(annotated_assemblies, assemblies)
+    if not len(assemblies):
+        logging.warning('All runs in this study are annotated with the latest pipeline.')
+        return
+
+    for assembly in assemblies:
+        assembly = mh.get_or_save_run(ena, study, assembly, args.lineage)
+        mh.create_annotation_job(request, assembly, args.priority)
+        logging.info('Created annotationJob for assembly {}'.format(assembly.primary_accession))
+
+
+def main(argv=None):
+    args = parse_args(argv)
+    if not args.annotate and not args.assemble:
+        logging.error('No job type specified, please set --annotate or --assemble in arugments.')
+        sys.exit(1)
+    mh = mgnify_handler.MgnifyHandler(args.db)
+    ena = ena_handler.EnaApiHandler()
+    try:
+        user = mh.get_user(args.webinID)
+    except ObjectDoesNotExist:
+        logging.warning('User {} not in db, creating user.'.format(args.webinID))
+        user_details = get_user_details(args.webinID)
+        user = mh.create_user(webin_id=args.webinID, email=user_details['email-address'],
+                              first_name=user_details['first-name'], surname=user_details['surname'],
+                              registered=True, consent_given=True)
+        logging.info('Created user {}'.format(args.webinID))
+
+    try:
+        request = mh.get_user_request(args.RTticket)
+        logging.warning('Request already exists according to RTticket number')
+    except ObjectDoesNotExist:
+        request = mh.create_user_request(user, args.priority, args.RTticket)
+
+    # Handle MGnify accessions
+    if 'MGYS' in args.study:
+        accession = get_study_secondary_accession(args.webinID, args.study)
+    # Handle ENA accessions
+    else:
+        accession = args.study
+
+    try:
+        if accession[0:3] in ('ERP', 'SRP', 'DRP'):
+            study = mh.get_backlog_secondary_study(accession)
+        else:
+            study = mh.get_backlog_study(accession)
+    except ObjectDoesNotExist:
+        try:
+            study_data = ena.get_study(accession)
+            study = mh.create_study_obj(study_data)
+        except json.decoder.JSONDecodeError:
+            raise ('Could not get study {} from ena.'.format(accession))
+        logging.info('Created study {}'.format(accession))
+
+    logging.info('Fetching study runs...')
```

Comment:
Please note that for each study we need to check for runs and assemblies. Unfortunately the ENA supports mixed studies, studies which consist of runs and assemblies. Here is an example (navigation tab!):
https://www.ebi.ac.uk/ena/data/view/PRJEB14154

Reply:
Resolved